### PR TITLE
Add filename sanitizing to adapters

### PR DESCRIPTION
**Pull request by masaball**
This PR moves filename sanitizing from the ffmpeg and passthrough adapters into a separate module. Additional sanitizing is added and applied in the two listed adapters as well as the AWS adapters. 

Sanitizing is strict, replacing any special characters outside the allowed set with `_` and removing all excess periods. The allowed set is `A-Z`, `a-z`, `0-9`, `.`, `-`, and `_`. This is a stricter version of the list provided by AWS for [object-keys](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html). Additionally, `/` is not sanitized because of how Avalon filenames are formatted when passed to Active Encode.

### CODE REVIEW

File: lib/active_encode/filename_sanitizer.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    end
+
+    def sanitize_base(input_url)
+      # Replace special characters with underscores and remove excess periods.
+      if input_url.is_a? URI::HTTP
+        # This removes the extension before processing so it is safe to delete all detected periods.
+        File.basename(input_url.path, File.extname(input_url.path)).gsub(/[^0-9A-Za-z.\-\/]/, '_').delete('.')
+      else
+        File.basename(input_url, File.extname(input_url)).gsub(/[^0-9A-Za-z.\-\/]/, '_').delete('.')
+      end
```
Comment (cjcolvar) [inline lib/active_encode/filename_sanitizer.rb:16]: The sanitization looks the same for both cases so how about creating a variable to hold the input?
```suggestion
      filepath = input_url.is_a?(URI::HTTP) ? input_url.path : input_url
      # This removes the extension before processing so it is safe to delete all detected periods.
      File.basename(filepath, File.extname(filepath)).gsub(/[^0-9A-Za-z.\-\/]/, '_').delete('.')
```

**Review by cjcolvar — COMMENTED** [review]
What do you think about making `ActiveEncode` be the path for calling the sanitization methods?

`ActiveEncode.sanitize_filename(input_url)`

This would avoid having to require the new module in all of the adapters and would make it easily available outside of ActiveEncode's codebase for use in apps that use ActiveEncode.  It would make method calls in adapters longer but I think it would avoid some bloat since the methods are generic and the adapters don't really need to provide those methods.

I think this is possible by adding the require to `lib/active_encode.rb` then declaring the module to extend the sanitization module:
```
module ActiveEncode
  extend ActiveEncode::FilenameSanitizer
end
```

File: lib/active_encode/engine_adapters/media_convert_adapter.rb [hunk 2]
```diff
@@ -482,6 +485,10 @@ def mediaconvert
         end
       end
 
+      def s3client
+        Aws::S3::Client.new
+      end
+
```
[the remark this answers is not in the document]
Comment (masaball) [reply lib/active_encode/engine_adapters/media_convert_adapter.rb:491]: It is referenced in the [upload_to_s3](https://github.com/samvera-labs/active_encode/blob/d4474f4e6b8f6b0ef00cde631ff7305740bbc7f2/lib/active_encode/engine_adapters/media_convert_adapter.rb#L516) method.

**Review by cjcolvar — COMMENTED** [review]
These changes look good to me, but I'm a little worried that there is the possibility that previously created encodes could be orphaned since this changes where ActiveEncode looks for files.  Do you think this could possibly happen?  If so, we could make this a major release due to the change in behavior.

**Comment by masaball** [comment]
I think the FileLocator is only really used by the engine adapters when doing the encoding and I don't think that previously created encodes reference it. But I may have overlooked something in the code or how external apps interact with it. So, I hope its not the case but it orphaning some previous encodes is a possibility.

**Review by cjcolvar — APPROVED** [review]
Looks great!  Thanks for your effort on this!

**Pull request merged**